backend/controllers/colaborador.js

```diff
@@ -104,24 +109,48 @@ export const inserirDadosColabCnpj = async(req,res)=>{
     }catch(error){
         await t.rollback()
         console.log(error)
-        res.json({message: error.message});
+        res.status(400).json({message: error.message});
     }
 }
 
 
 export const geralFunc = async (req,res) =>{
     try{
-        const dados = await sequelize.query(
-        `select c.nome, ca.cargo, c.email, c.telefone, a.area 
-            from Colaboradors c 
-        left join Cargos as ca 
-            on c.Cargos_ID = ca.ID
-        left join Departamentos as a 
-            on a.ID = ca.Departamento_ID;`,
-        {type:sequelize.QueryTypes.SELECT})
+        const dados = await Colaborador.findAll({
+            attributes:['nome','telefone','ID','email'],
+            include:{
+                model:Cargo,
+                attributes:['cargo','ID'],
+                include:{
+                    model:Departamento,
+                    attributes:['area','ID']
+                },
+            }
+        });
         res.json({dados})
     }
     catch(error){
+        res.status(500).json({message:error.message})
+    }
+}
+
+export const getCargoColaborador = async (req,res) =>{
+    try{ 
+        const cargo_colab = await Colaborador.findAll({
+            include:{
+                model:Cargo,
+                include:{
+                    model:Departamento,
+                    where:{
+                        ID:req.params.id
+                    },
+                    required:true
+                },
+                required:true
+            }
+        });
+        res.json(cargo_colab)
+    }catch(error){
         res.json({message:error.message})
```

Comment:
Adicionar quando possível o status do erro  
```suggestion

     res.status(400).json({message:error.message})
```